### ISSUE TITLE
Invalid TabIndex causing accessibility test failures

### ISSUE THREAD

**Issue by BarbaraPruz** [wojtekmaj/react-pdf#1913]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [X] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

We are trying to ensure our app is accessible. For part of our evaluation, we use axe devtools to do full page scans. It is reporting an error that elements should not have tabindex greater than 0. Here is link to error description: https://dequeuniversity.com/rules/axe/4.10/tabindex?application=AxeChrome

Corresponding Element Inspection:
```
<div class="react-pdf__Page__annotations annotationLayer" data-main-rotation="0" style="width: round(var(--scale-factor) * 612px, 1px); height: round(var(--scale-factor) * 792px, 1px);">
  <section data-annotation-id="13R" tabindex="1000" class="linkAnnotation" style="z-index: 0; left: 21.0784%; top: 95.9183%; width: 13.3578%; height: 1.23106%;">
    <a data-element-id="13R" href="http://passport.steadyapp.com/" rel="noopener noreferrer nofollow" target="" id="pdfjs_internal_id_13R"></a>
  </section>
  <section data-annotation-id="14R" tabindex="1000" class="linkAnnotation" style="z-index: 1; left: 39.7059%; top: 95.9183%; width: 14.2157%; height: 1.23106%;">
    <a data-element-id="14R" href="mailto:passport@steadyapp.com" rel="noopener noreferrer nofollow" target="" id="pdfjs_internal_id_14R">
    </a>
  </section>
</div>
```

### Steps to reproduce

If useful, I can provide a sample pdf. But guessing this is related to links in general.

### Expected behavior

react-pdf viewer page passes WCAG AA requirements including valid tab index values.

### Actual behavior

tabindex is set to 1000

### Environment

- **Browser (if applicable)**: Chrome Version 130.0.6723.92 (Official Build) (arm64)
- **React-PDF version**: 9.1.1
- **React version**: 18.0.2
- **Bundler name and version (if applicable)**:
